core/src/main/java/org/apache/calcite/sql/SqlDataTypeSpec.java

```diff
@@ -162,7 +161,7 @@ public TimeZone getTimeZone() {
   /** Returns a copy of this data type specification with a given
    * nullability. */
   public SqlDataTypeSpec withNullable(Boolean nullable) {
-    if (SqlFunctions.eq(nullable, this.nullable)) {
+    if (nullable.equals(this.nullable)) {
```

Comment:
It's only called in one place, from the parser, with argument 'false', so it doesn't matter too much. But let's do it properly: call `java.util.Objects.equals(Object, Object)`.